src/main/java/me/lucien/minesweeper/domain/RoomManager.java

```diff
@@ -0,0 +1,113 @@
+package me.lucien.minesweeper.domain;
+
+import me.lucien.minesweeper.web.HttpException;
+import org.json.JSONObject;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+@Component
+public class RoomManager {
+
+    @Value("${room.capacity}")
+    private int capacity;
+
+    @Value("${room.delay}")
+    private int delay;
+
+    private Map<Integer, Room> roomMap = new HashMap<>();
+    private Map<Integer, ScheduledFuture> taskMap = new HashMap<>();
+    private ScheduledExecutorService executorService = Executors.newSingleThreadScheduledExecutor();
+
+    public String createRoom(int width, int height) throws HttpException {
+        if (roomMap.size() >= capacity) {
+            throw new HttpException(HttpStatus.SERVICE_UNAVAILABLE, "No available room.");
+        }
+
+        Room room = new Room(width, height);
+        roomMap.put(room.getId(), room);
+
+        ScheduledFuture future = executorService.schedule(
+                () -> { roomMap.remove(room.getId()); }, delay, TimeUnit.SECONDS);
+        taskMap.put(room.getId(), future);
+
+        JSONObject o = new JSONObject();
+        o.put("id", room.getId());
+        o.put("key", room.getKey());
+        o.put("mineNum", room.getMineNum());
+
+        return o.toString();
+    }
+
+    public List<SquareData> enterRoom(int id, String key) throws HttpException {
```

Comment:
There should not be `enterRoom()`. Its behavior and methods should be like a collection or a container. Do not return List<SquareData>. It manages room instances only. Find and return a room.